lib/ansible/modules/network/fortios/fortios_voip_profile.py

```diff
@@ -0,0 +1,1152 @@
+#!/usr/bin/python
+from __future__ import (absolute_import, division, print_function)
+# Copyright 2019 Fortinet, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'metadata_version': '1.1'}
+
+DOCUMENTATION = '''
+---
+module: fortios_voip_profile
+short_description: Configure VoIP profiles in Fortinet's FortiOS and FortiGate.
+description:
+    - This module is able to configure a FortiGate or FortiOS by allowing the
+      user to set and modify voip feature and profile category.
+      Examples include all parameters and values need to be adjusted to datasources before usage.
+      Tested with FOS v6.0.2
+version_added: "2.8"
+author:
+    - Miguel Angel Munoz (@mamunozgonzalez)
+    - Nicolas Thomas (@thomnico)
+notes:
+    - Requires fortiosapi library developed by Fortinet
+    - Run as a local_action in your playbook
+requirements:
+    - fortiosapi>=0.9.8
+options:
+    host:
+       description:
+            - FortiOS or FortiGate ip address.
+       required: true
+    username:
+        description:
+            - FortiOS or FortiGate username.
+        required: true
+    password:
+        description:
+            - FortiOS or FortiGate password.
+        default: ""
+    vdom:
+        description:
+            - Virtual domain, among those defined previously. A vdom is a
+              virtual instance of the FortiGate that can be configured and
+              used as a different unit.
+        default: root
+    https:
+        description:
+            - Indicates if the requests towards FortiGate must use HTTPS
+              protocol
+        type: bool
+        default: true
+    voip_profile:
+        description:
+            - Configure VoIP profiles.
+        default: null
+        suboptions:
+            state:
```

Comment:
Ideally `state` option shall be used under module _(i.e. fortios_voip_profile)_, so wanted to verify if here other than `voip_profile` param `fortios_voip_profile` supports other options as well or its planned in future release, if not then it would be better to use `state` option in conjunction with the module

Reply:
Yes, actually it is planned for the future and other modules already do it: https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/network/fortios/fortios_webfilter.py. There are two 'submodules': webfilter_url and webfilter_content

Reply:
@mamunozgonzalez so, I believe you shall be doing the respective changes in Ansible future release not in `2.8`?

Reply:
Well, if we are on time we would like to improve the code as much as possible before 2.8 to ensure the best user experience. We are putting more efforts now to develop and submit the modules so we are trying our best to do it.

We see that many customers are starting to use these modules (and we are getting good feedback and traction) thus we will put more efforts into improving this as much as possible, now and in the future.

Reply:
OK, please take a pointer of the same and if at all feasible then I would suggest to cram in `2.8`, as coming back to address the issue is always hectic n sometimes forgotten I believe.